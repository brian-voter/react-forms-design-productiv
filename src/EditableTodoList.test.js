import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList.js";
import { initialTodos } from "./_testCommon.js";
const update = jest.fn();
const remove = jest.fn();

describe("EditableTodoList Component smoke and snapshot", function () {
  test("smoke test - renders without crashing", function () {
    render(<EditableTodoList todos={initialTodos} />);
  });

  test("snapshot test", function () {
    const { container } = render(
      <EditableTodoList
        todos={initialTodos} />);
    expect(container).toMatchSnapshot();
  });
});

describe("EditableTodoList renders correct todos", function () {
  test("EditableTodoList renders correct todos", function () {
    const { container, queryByText } = render(<EditableTodoList todos={initialTodos} />);

    for (const { title, description } of initialTodos) {
      expect(queryByText(title)).toBeInTheDocument();
      expect(queryByText(description)).toBeInTheDocument();
    }
  });
});

describe("EditableTodoList buttons trigger parent handler functions correctly", function () {
  test("EditableTodoList calls remove correctly", function () {
    const remove = jest.fn();

    const { container, queryByText } = render(
      <EditableTodoList
        todos={initialTodos}
        remove={remove}
      />);

    const deleteBtns = container.querySelectorAll("#delete-btn");

    expect(deleteBtns.length).toEqual(initialTodos.length);

    for (let i = 0; i < deleteBtns.length; i++) {
      const { id } = initialTodos[i];
      fireEvent.click(deleteBtns[i]);

      // ensure remove was called with the id of this todo
      expect(remove.mock.lastCall[0]).toBe(id);
    }
  });

  test("EditableTodoList calls update correctly", function () {
    const remove = jest.fn();

    const { container, queryByText } = render(
      <EditableTodoList
        todos={initialTodos}
        remove={remove}
      />);

    const deleteBtns = container.querySelectorAll("#delete-btn");

    expect(deleteBtns.length).toEqual(initialTodos.length);

    for (let i = 0; i < deleteBtns.length; i++) {
      const { id } = initialTodos[i];
      fireEvent.click(deleteBtns[i]);

      // ensure remove was called with the id of this todo
      expect(remove.mock.lastCall[0]).toBe(id);
    }
  });
});