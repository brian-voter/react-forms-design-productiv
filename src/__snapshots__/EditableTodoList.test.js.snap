// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodoList Component smoke and snapshot snapshot test 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Code!
            </b>
             
            <small>
              (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              Write some code
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Make dinner
            </b>
             
            <small>
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              Cook something healthy
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Go to bed
            </b>
             
            <small>
              (priority: 
              3
              )
            </small>
          </div>
          <div>
            <small>
              In bed by 11:15
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EditableTodoList Component smoke and snapshot snapshot test 2`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Code!
            </b>
             
            <small>
              (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              Write some code
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Make dinner
            </b>
             
            <small>
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              Cook something healthy
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Go to bed
            </b>
             
            <small>
              (priority: 
              3
              )
            </small>
          </div>
          <div>
            <small>
              In bed by 11:15
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EditableTodoList renders  snapshot test 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Code!
            </b>
             
            <small>
              (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              Write some code
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Make dinner
            </b>
             
            <small>
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              Cook something healthy
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Go to bed
            </b>
             
            <small>
              (priority: 
              3
              )
            </small>
          </div>
          <div>
            <small>
              In bed by 11:15
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EditableTodoList renders correct todos snapshot test 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Code!
            </b>
             
            <small>
              (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              Write some code
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Make dinner
            </b>
             
            <small>
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              Cook something healthy
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-btn"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            id="delete-btn"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Go to bed
            </b>
             
            <small>
              (priority: 
              3
              )
            </small>
          </div>
          <div>
            <small>
              In bed by 11:15
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
